lib/livebook/hubs/team_client.ex

```diff
@@ -397,6 +406,34 @@ defmodule Livebook.Hubs.TeamClient do
     |> dispatch_deployment_groups(agent_connected)
   end
 
+  defp handle_event(:agent_key_created, agent_key_created, state) do
+    agent_key = build_agent_key(agent_key_created)
+
+    case find_deployment_group(agent_key.deployment_group_id, state.deployment_groups) do
+      nil ->
+        state
+
+      deployment_group ->
+        agent_keys = Enum.uniq([agent_key | deployment_group.agent_keys])
+        deployment_group = %{deployment_group | agent_keys: Enum.reverse(agent_keys)}
+        handle_event(:deployment_group_updated, deployment_group, state)
+    end
+  end
+
+  defp handle_event(:agent_key_deleted, agent_key_deleted, state) do
+    agent_key = build_agent_key(agent_key_deleted)
+
+    case find_deployment_group(agent_key.deployment_group_id, state.deployment_groups) do
+      nil ->
+        state
+
+      deployment_group ->
+        agent_keys = Enum.reject(deployment_group.agent_keys, &(&1 == agent_key))
```

Comment:
```suggestion
        agent_keys = Enum.reject(deployment_group.agent_keys, &(&1.id == agent_key.id))
```

Sidenote: we could `List.delete/2`, but when there are ids we can compare that's usually a better default.